plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1800,6 +1795,13 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
             }
         }
 
+        if (properties.containsKey(EXTRA_PROPERTIES)) {
+            Map<String, String> extraProperties = (Map<String, String>) properties.get(EXTRA_PROPERTIES)
+                    .orElseThrow(() -> new IllegalArgumentException("extra_properties property cannot be empty"));
+
+            extraProperties.forEach(updateProperties::set);
+        }
+
```

Comment:
Currently this doesn't work.
This part needs to move to somewhere before `updateProperties.commit();` (line 1771).

```
trino> ALTER TABLE test.property_test SET PROPERTIES extra_properties = MAP(ARRAY['write.metadata.delete-after-commit.enabled'], ARRAY['true']);
SET PROPERTIES
trino> SELECT * FROM test."property_test$properties";
               key               | value
---------------------------------+-------
 write.format.default            | ORC
 write.parquet.compression-codec | zstd
(2 rows)
```

Reply:
https://github.com/trinodb/trino/pull/17595#discussion_r1293333030
> To avoid such situations we should disallow table property names handled by Trino.

According to the discussion above, table properties which can be set by other features shouldn't be able to set by `extra_properties`.

Please refer to the following implementation:
https://github.com/trinodb/trino/pull/17595/files#diff-5f2b7463951afa38061e5a1329d04fe267759718d2201597e1e578d8f83ec446R180-R189

Reply:
Fixed!